core/dbt/contracts/graph/parsed.py

```diff
@@ -221,8 +224,43 @@ class ParsedNodeDefaults(NodeInfoMixin, ParsedNodeMandatory):
     unrendered_config: Dict[str, Any] = field(default_factory=dict)
     created_at: float = field(default_factory=lambda: time.time())
     config_call_dict: Dict[str, Any] = field(default_factory=dict)
+    constraints_enabled: bool = False
+
+    def constraints_validator(self):
+        materialization_error = {}
+        language_error = {}
+        data_type_errors = set()
+
+        if self.resource_type == "model" and self.config.constraints_enabled is True:
+            if self.config.materialized != "table":
```

Comment:
Should this be so restrictive to `table`? 

Some folks may be bold enough to create their own `incremental` materializations based on the the new configs available.

Reply:
Let's leave it alone as `table` only for now and see what can be done for `incremental` after we get more user feedback